scripts/deploy.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+# ideas used from https://gist.github.com/motemen/8595451
+
+# abort the script if there is a non-zero error
+set -e
+
+# show where we are on the machine
+pwd
+
+remote=$(git config remote.origin.url)
+
+siteSource="$1"
+
+if [ ! -d "$siteSource" ]
+then
+    echo "Usage: $0 <site source dir>"
+    exit 1
+fi
+
+# make a directory to put the gp-pages branch
+mkdir gh-pages-branch
+cd gh-pages-branch
+# now lets setup a new repo so we can update the gh-pages branch
+git config --global user.email "$GH_EMAIL" > /dev/null 2>&1
+git config --global user.name "$GH_NAME" > /dev/null 2>&1
+git init
+git remote add --fetch origin "$remote"
+
+# switch into the the gh-pages branch
+if git rev-parse --verify origin/gh-pages > /dev/null 2>&1
+then
+    git checkout gh-pages
+    # delete any old site as we are going to replace it
+    # Note: this explodes if there aren't any, so moving it here for now
+    git rm -rf .
```

Comment:
this looks scary! 😨 

I guess it will always be in `gh-pages-branch` directory...